tiledb/common/interval/interval.h

```diff
@@ -964,6 +964,7 @@ class Interval : public detail::IntervalBase {
       }
     }
     if constexpr (!Traits::has_infinite_elements) {
+      (void)is_for_upper_bound;
```

Comment:
If VS2017 is warning about an unreferenced variable here, it's indeed a defect in VS2017.

I don't mind some work-around, but we need a work-around that can be reverted easily. I really don't want to leave an uncommented voided value in the code that works around a particular defect in a soon-to-be-unsupported compiler. A macro definition with a unique name is called for here, so that when we drop support for that compiler, we can also eliminate the macro and all its usages in a single PR.

While this is at present the only such problem with `if  constexpr`, it won't be the last one for the duration where we need to support VS2017. The same work-around will be needed in other files. I'd call the header `compiler_hacks.h`, partly because I doubt this is the last hack of this kind we'll have to deal with, and partly because, in a new branch, I've got `macro_hacks_begin.h` and `macro_hacks_end.h` to isolate some reusable preprocessor nonsense that's similar to what's needed here.